presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarFromAnnotationsParser.java

```diff
@@ -99,11 +107,38 @@ private static List<ScalarHeaderAndMethods> findScalarsInFunctionSetClass(Class<
         return builder.build();
     }
 
+    private static Optional<ScalarStatsHeader> getScalarStatsHeader(Method annotated)
+    {
+        Optional<ScalarStatsHeader> scalarStatsHeader;
+        ScalarFunctionConstantStats constantStatsAnnotation =
+                annotated.getAnnotation(ScalarFunctionConstantStats.class);
+        List<java.lang.reflect.Parameter> params =
+                Arrays.stream(annotated.getParameters())
+                        .filter(param -> param.getAnnotation(SqlType.class) != null)
+                        .collect(Collectors.toList());
+        // Map of (function argument position index) -> (ScalarPropagateSourceStats annotation)
+        ImmutableMap.Builder<Integer, ScalarPropagateSourceStats> argumentStatsResolverBuilder = new ImmutableMap.Builder<>();
```

Comment:
This can be simplified to just a `List<ScalarPropagateSourceStats>`, the list index is the argument position index

Reply:
Challenge here is, what do I add at those argument position where we do not have an annotation. 
```java
if (params.get(paramIndex).getAnnotation(ScalarPropagateSourceStats.class) == null) {
                argumentStatsResolverBuilder.add((ScalarPropagateSourceStats) null);
            }
```
This code produces null pointer exception with Immutable list. 

* One option is to use Optional.valueOf, but that will make it a List<Optional<ScalarPropagateSourceStats>>..

Reply:
Ok, makes sense then